src/main/java/io/cdap/plugin/servicenow/source/ServiceNowBaseSourceConfig.java

```diff
@@ -163,39 +163,36 @@ private void validateValueType(FailureCollector collector) {
 
   private void validateDateRange(FailureCollector collector) {
     if (containsMacro(ServiceNowConstants.PROPERTY_START_DATE) ||
-      containsMacro(ServiceNowConstants.PROPERTY_END_DATE)) {
+      containsMacro(ServiceNowConstants.PROPERTY_END_DATE) ||
+      (Util.isNullOrEmpty(startDate) && Util.isNullOrEmpty(endDate))) {
       return;
     }
 
-    if (!Util.isNullOrEmpty(startDate) && !Util.isNullOrEmpty(endDate) &&
-      !Util.isValidDateFormat(ServiceNowConstants.DATE_FORMAT, startDate) &&
-      !Util.isValidDateFormat(ServiceNowConstants.DATE_FORMAT, endDate)) {
-      collector.addFailure("Invalid format for Start date. Correct Format: " +
-          ServiceNowConstants.DATE_FORMAT, null)
+    if (Util.isNullOrEmpty(startDate) && !Util.isNullOrEmpty(endDate)) {
```

Comment:
as mentioned before, these checks should be moved down after the date format checks. That way, if an invalid end date is given without a start date, both problems are highlighted at the same time. It would also be more readable to structure it like:

```
if (Util.isNullOrEmpty(endDate)) {
  if (Util.isNullOrEmpty(startDate)) {
    add failure
  }
  if (end date is not valid) {
    add failure
  }
}

if (Util.isNullOrEmpty(startDate)) {
  if (Util.isNullOrEmptry(endDate)) {
    add failure
  }
  if (start date is not valid) {
    add failure
  }
}
```

Reply:
Refactored the code as per the above suggestion. PTAL. f51e053a9263b0190dc1d3408b82c5de716af698